tests/test_ptcDataset.py

```diff
@@ -61,84 +62,177 @@ def setUp(self):
-                                                                           nSideCovMatrix), 10.0).tolist()
-                    localDataset.aMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), 1e-6).tolist()
-                    localDataset.bMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), 1e-7).tolist()
+                                                                           nSideCovMatrix), 10.0)
+                    localDataset.aMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), 1e-6)
+                    localDataset.bMatrix[ampName] = np.full((nSideCovMatrix, nSideCovMatrix), 1e-7)
                     localDataset.covariancesModelNoB[ampName] = np.full((nSignalPoints, nSideCovMatrix,
-                                                                        nSideCovMatrix), 15.0).tolist()
+                                                                        nSideCovMatrix), 15.0)
                     localDataset.aMatrixNoB[ampName] = np.full(
-                        (nSideCovMatrix, nSideCovMatrix), 2e-6).tolist()
+                        (nSideCovMatrix, nSideCovMatrix), 2e-6)
 
-            filename = tempfile.mktemp()
-            usedFilename = localDataset.writeText(filename + ".yaml")
-            fromText = PhotonTransferCurveDataset.readText(usedFilename)
+            with tempfile.NamedTemporaryFile(suffix=".yaml") as f:
+                usedFilename = localDataset.writeText(f.name)
+                fromText = PhotonTransferCurveDataset.readText(usedFilename)
             self.assertEqual(localDataset, fromText)
 
-            filename = tempfile.mktemp()
-            usedFilename = localDataset.writeFits(filename + ".fits")
-            fromFits = PhotonTransferCurveDataset.readFits(usedFilename)
+            with tempfile.NamedTemporaryFile(suffix=".fits") as f:
+                usedFilename = localDataset.writeFits(f.name)
+                fromFits = PhotonTransferCurveDataset.readFits(usedFilename)
             self.assertEqual(localDataset, fromFits)
 
+    def test_getExpIdsUsed(self):
+        localDataset = copy.copy(self.dataset)
+
+        for pair in [(12, 34), (56, 78), (90, 10)]:
+            localDataset.inputExpIdPairs["C:0,0"].append(pair)
+        localDataset.expIdMask["C:0,0"] = np.array([True, False, True])
+        self.assertTrue(np.all(localDataset.getExpIdsUsed("C:0,0") == [(12, 34), (90, 10)]))
+
+        localDataset.expIdMask["C:0,0"] = np.array([True, False, True, True])  # wrong length now
+        with self.assertRaises(AssertionError):
+            localDataset.getExpIdsUsed("C:0,0")
+
+    def test_getGoodAmps(self):
+        dataset = self.dataset
+
+        self.assertTrue(dataset.ampNames == self.ampNames)
+        dataset.badAmps.append("C:0,1")
+        self.assertTrue(dataset.getGoodAmps() == [amp for amp in self.ampNames if amp != "C:0,1"])
+
+    def test_ptcDataset_pre_dm38309(self):
```

Comment:
Do we need an analogous test for the partial PTC datasets created by `cpPtcExtract`?

Reply:
Those are fine, in that the partial datasets didn't have the extra dimension.  But it's also not something that is a common use-case anyway as those are not really user-facing.